neuraxle/metaopt/trial.py

```diff
@@ -50,32 +55,224 @@ def __init__(
             self,
             hyperparams: HyperparameterSamples,
             status: 'TRIAL_STATUS' = None,
+            pipeline: BaseStep = None,
+            validation_splits=None,
+            cache_folder=None,
+            error=None,
+            error_traceback=None,
+            start_time=None,
+            end_time=None
+    ):
+        if status is None:
+            status = TRIAL_STATUS.PLANNED
+        if validation_splits is None:
+            validation_splits = []
+
+        self.status: TRIAL_STATUS = status
+        self.hyperparams = hyperparams
+        self.pipeline: BaseStep = pipeline
+        self.validation_splits = validation_splits
+        self.cache_folder = cache_folder
+        self.error_traceback = error_traceback
+        self.error = error
+        self.start_time = start_time
+        self.end_time = end_time
+
+    def new_validation_split(self, pipeline: BaseStep):
+        """
+        Create a new trial split.
+        A trial has one split when the validation splitter function is validation split.
+        A trial has one or many split when the validation splitter function is kfold_cross_validation_split.
+
+        :type pipeline: pipeline to execute
+        :return:
+        """
+        trial_split: TrialSplit = TrialSplit(split_number=len(self.validation_splits))
```

Comment:
```suggestion
        trial_split: TrialSplit = TrialSplit(total_split_count=len(self.validation_splits))
```
?

Reply:
our discussion gave me an idea : https://github.com/Neuraxio/Neuraxle/issues/297
resolved this to issue. split_number is okey for now